apiserver/facades/client/application/deployrepository.go

```diff
@@ -806,36 +815,36 @@ func (v *deployFromRepositoryValidator) getCharm(arg params.DeployFromRepository
 	// be populated once the charm gets downloaded.
 	resolvedData, err := v.resolveCharm(initialCurl, requestedOrigin, arg.Force, usedModelDefaultBase, arg.Cons)
 	if err != nil {
-		return nil, corecharm.Origin{}, nil, err
+		return "", corecharm.Origin{}, nil, err
 	}
 	resolvedOrigin := resolvedData.EssentialMetadata.ResolvedOrigin
 	deployRepoLogger.Tracef("from resolveCharm: %s, %s", resolvedData.URL, pretty.Sprint(resolvedOrigin))
 	if resolvedOrigin.Type != "charm" {
-		return nil, corecharm.Origin{}, nil, errors.BadRequestf("%q is not a charm", arg.CharmName)
+		return "", corecharm.Origin{}, nil, errors.BadRequestf("%q is not a charm", arg.CharmName)
 	}
 
 	resolvedCharm := corecharm.NewCharmInfoAdapter(resolvedData.EssentialMetadata)
 	if resolvedCharm.Meta().Name == bootstrap.ControllerCharmName {
-		return nil, corecharm.Origin{}, nil, errors.NotSupportedf("manual deploy of the controller charm")
+		return "", corecharm.Origin{}, nil, errors.NotSupportedf("manual deploy of the controller charm")
 	}
 
 	// Check if a charm doc already exists for this charm URL. If so, the
 	// charm has already been queued for download so this is a no-op. We
 	// still need to resolve and return back a suitable origin as charmhub
 	// may refer to the same blob using the same revision in different
 	// channels.
-	deployedCharm, err := v.state.Charm(resolvedData.URL)
+	deployedCharm, err := v.state.Charm(resolvedData.URL.String())
 	if err != nil && !errors.Is(err, errors.NotFound) {
-		return nil, corecharm.Origin{}, nil, errors.Trace(err)
+		return "", corecharm.Origin{}, nil, errors.Trace(err)
 	} else if err == nil {
-		return resolvedData.URL, resolvedOrigin, deployedCharm, nil
+		return resolvedData.URL.String(), resolvedOrigin, deployedCharm, nil
 	}
 
 	// This charm needs to be downloaded, remove the ID and Hash to
 	// allow it to happen.
 	resolvedOrigin.ID = ""
 	resolvedOrigin.Hash = ""
-	return resolvedData.URL, resolvedOrigin, resolvedCharm, nil
+	return resolvedData.URL.String(), resolvedOrigin, resolvedCharm, nil
```

Comment:
Because ideally in the next step (or one of the next steps) I wouldn't need to parse the url to get the revision, or to construct a charm.ID

However, as with an above comment, this can be done when this is implemented